app/models/assignment_participant.rb

```diff
@@ -53,6 +55,31 @@ def review_score
   def scores(questions)
     scores = {}
     scores[:participant] = self
+    assignment_questionnaires(questions, scores)
+    scores[:total_score] = self.assignment.compute_total_score(scores)
+    # merge scores[review#] (for each round) to score[review]  -Yang
+    merge_scores(scores) if self.assignment.varying_rubrics_by_round?
+    # In the event that this is a microtask, we need to scale the score accordingly and record the total possible points
+    # PS: I don't like the fact that we are doing this here but it is difficult to make it work anywhere else
+    topic_total_scores(scores) if self.assignment.is_microtask?
+
+    # for all quiz questionnaires (quizzes) taken by the participant
+    # quiz_responses = []
+    # quiz_response_mappings = QuizResponseMap.where(reviewer_id: self.id)
+    # quiz_response_mappings.each do |qmapping|
+    #   quiz_responses << qmapping.response if qmapping.response
+    # end
+    # scores[:quiz] = Hash.new
+    # scores[:quiz][:assessments] = quiz_responses
+    # scores[:quiz][:scores] = Answer.compute_quiz_scores(scores[:quiz][:assessments])
+    scores[:total_score] = assignment.compute_total_score(scores)
+    # scores[:total_score] += compute_quiz_scores(scores)
+    # move lots of calculation from view(_participant.html.erb) to model
+    caculate_scores(scores)
+  end
+
+  # methods extracted from scores method:assignment_questionnaires, merge_scores, topic_total_scores, caculate_scores
```

Comment:
Good comment! You could move this comment before the definition of `scores` method.